src/asset-services.ts

```diff
@@ -0,0 +1,123 @@
+function ensureAbsoluteURLs(baseURL: string, entrypoints: string[]) {
+  const regExpIsAbsoluteURL = new RegExp("^(?:[a-z]+:)?//", "i");
+  return entrypoints.map(function (entrypoint) {
+    if (!regExpIsAbsoluteURL.test(entrypoint)) {
+      return baseURL + entrypoint;
+    }
+    return entrypoint;
+  });
+}
+
+function addRef(document: Document, entrypoint: string) {
+  const pattern = /\.([0-9a-z]+)(?=[?#])|(\.)(?:[\w]+)$/gim;
+  switch (entrypoint.match(pattern)![0]) {
+    case ".css":
+      let link = document.createElement("link");
+      link.rel = "stylesheet";
+      link.href = entrypoint;
+      (link as any).async = false; // TODO: it seems to be superfluous, remove
+      // TODO: DE-666 - render this element near the script element where load is invoked
+      document.head.appendChild(link);
+      break;
+    case ".js":
+      let script = document.createElement("script");
+      script.src = entrypoint;
+      script.async = false;
+      // TODO: DE-666 - render this element near the script element where load is invoked
+      document.body.appendChild(script);
+      break;
+    default:
+      break;
+  }
+}
+
+async function load({
+  fetch,
+  document,
+  manifestURL,
+  sources,
+}: {
+  fetch: (input: RequestInfo, init?: RequestInit) => Promise<Response>;
+  document: Document;
+  manifestURL: string;
+  sources: string[];
+}): Promise<void> {
+  try {
+    const response = await fetch(manifestURL);
+    const data = await response.json();
+    const entrypoints = ensureAbsoluteURLs(
+      manifestURL.substring(0, manifestURL.lastIndexOf("/") + 1),
+      data.entrypoints
+    );
+    entrypoints.concat(sources).forEach((entrypoint) => {
+      addRef(document, entrypoint);
+    });
+  } catch (error) {
+    // TODO: DE-667 - improve it, we are loosing error information here
+    // Consider using Loggly
+    // Consider applying retries
+    // Consider allowing run fallback code
+    throw new Error("Error getting assets file: " + manifestURL);
+  }
+}
+
+function parseArgs(
```

Comment:
Quedo muy bueno esto para la retro compatibilidad pero el nombre no me dice que es para la retro compatibilidad, podríamos buscar un nombre mas diciente?

```suggestion
function argsParserToRetrocompatibility(
```

Reply:
Mmm, no estoy seguro de que sea solo para retro compatibilidad, esta función también se encargaría de los valores por defecto, etc.

¿Que opinás de?

```suggestion
function normalizeArgs(
```
 
Otra opción es quitarla. Ya que el _load_ de la clase ahora simplemente _parsea_ los parámetros y luego llama al _load_ estático. Osea, podría hacer esto directamente dentro del _load_ de la clase.

Reply:
Ok, revirtiendo mi cambio jeje